tools/cli/src/mentat_cli/command_parser.rs

```diff
@@ -62,18 +65,44 @@ pub fn command(s: &str) -> Command {
                     });
 
     let close_parser = string(CLOSE_COMMAND)
-                    .and(skip_many(space()))
+                    .and(skip_spaces.clone())
                     .map( |_| Command::Close );
 
-    skip_many(space())
+    let opening_brace_parser = try(string("["))
+                    .or(try(string("{")));
+    
+    let edn_arg_parser = skip_spaces.clone()
+                    .and(opening_brace_parser.clone()
+                    .and(many::<Vec<_>, _>(try(any()))));
```

Comment:
Can you explain why you need to `try` here?  `any` will either consume and succeed or not consume and fail; there shouldn't be consume and fail.